pkg/services/ngalert/store/provisioning_store.go

```diff
@@ -0,0 +1,81 @@
+package store
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/grafana/grafana/pkg/services/ngalert/models"
+	"github.com/grafana/grafana/pkg/services/sqlstore"
+)
+
+type provenanceRecord struct {
+	Id         int `xorm:"pk autoincr 'id'"`
+	OrgId      int
+	RecordKey  string
+	RecordType string
+	Provenance models.Provenance
+}
+
+func (pr provenanceRecord) TableName() string {
+	return "provenance_type"
+}
+
+type ProvisioningStore interface {
+	GetProvenance(o models.ProvisionedObject) (models.Provenance, error)
```

Comment:
I think we should also add the `orgID` as part of the parameters, as we want to filter by org. Also `context.Context` as George did mention.

Reply:
Ahhh, yep, good call. This PR doesn't add any tests which cover this, either.

Reply:
I wonder if we should consider instead reading the orgID off the provided object if the field exists? Any domain object that respects organizations already has the field, and it feels cleaner to work in terms of the DTO purely. (I.e. extending the `Provisionable` interface with a third method to read the org ID)

Reply:
Tried this for alert rules and I like how it came out. I'll move with this for now. If we decide we don't like it on any of the future types we can always change it.

Also, thanks for calling this out, as the unique index added by the migration was missing the org_id column. It was much easier to add it now rather than fix the index in a future PR!